From 0f07dadba882760201e44de0d63bd440c987f8c2 Mon Sep 17 00:00:00 2001
From: Samuel Kirby <sk5112@ic.ac.uk>
Date: Fri, 26 Feb 2021 02:04:38 +0000
Subject: [PATCH 3/3] notify controller on exit

---
 src/SmileiMPI/SmileiMPI.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/SmileiMPI/SmileiMPI.cpp b/src/SmileiMPI/SmileiMPI.cpp
index 430af1a..60a6311 100755
--- a/src/SmileiMPI/SmileiMPI.cpp
+++ b/src/SmileiMPI/SmileiMPI.cpp
@@ -75,6 +75,13 @@ SmileiMPI::~SmileiMPI()
 {
     delete[]periods_;
 
+    MPI_Comm controller;
+    MPI_Comm_get_parent(&controller);
+    if (controller != MPI_COMM_NULL) {
+        int8_t buf = 0;
+        MPI_Send(&buf, 1, MPI_INT8_T, 0, 0, controller);
+    }
+
     MPI_Finalize();
 
 } // END SmileiMPI::~SmileiMPI
-- 
1.8.3.1

