From f72dfc9a19924940411663636373941fb33098fb Mon Sep 17 00:00:00 2001
From: Sam Kirby <sam.kirby94@hotmail.co.uk>
Date: Sun, 28 Feb 2021 11:18:15 +0000
Subject: [PATCH 3/4] notify controller on exit

---
 src/SmileiMPI/SmileiMPI.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/SmileiMPI/SmileiMPI.cpp b/src/SmileiMPI/SmileiMPI.cpp
index 430af1ae5..b9755b0fc 100755
--- a/src/SmileiMPI/SmileiMPI.cpp
+++ b/src/SmileiMPI/SmileiMPI.cpp
@@ -75,6 +75,16 @@ SmileiMPI::~SmileiMPI()
 {
     delete[]periods_;
 
+    MPI_Comm controller;
+    MPI_Comm_get_parent(&controller);
+
+    if (controller != MPI_COMM_NULL) {
+        MPI_Request req;
+        MPI_Ibarrier(controller, &req);
+        MPI_Wait(&req, MPI_STATUS_IGNORE);
+        MPI_Comm_disconnect(&controller);
+    }
+
     MPI_Finalize();
 
 } // END SmileiMPI::~SmileiMPI
-- 
2.30.1

